syntax = "proto3";

package transport_catalogue_proto;

import "map_render.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    uint32 id = 1;
    string name = 2;
    Coordinates coordinates = 3;
}

message Bus {
    uint32 id = 1;
    string name = 2;
    repeated uint32 stop_ids = 3;
    bool is_circle = 4;
}

message StopToStopDistance {
    uint32 from_stop_id = 1;
    uint32 to_stop_id = 2;
    double distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopToStopDistance stop_to_stop_distances = 3;
}

message Catalogue {
    TransportCatalogue transport_catalogue = 1;
    RenderSettings render_settings = 2;
}